public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -69,18 +136,58 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   } = useFormContext<RuleFormValues>();
 
   const { queryPreviewData, runQueries, cancelQueries, isPreviewLoading, clearPreviewData } = useAlertQueryRunner();
+  const isSwitchModeEnabled = config.featureToggles.alertingQueryAndExpressionsStepMode ?? false;
 
   const initialState = {
     queries: getValues('queries'),
   };
 
   const [{ queries }, dispatch] = useReducer(queriesAndExpressionsReducer, initialState);
-  const [type, condition, dataSourceName] = watch(['type', 'condition', 'dataSourceName']);
+
+  // data queries only
+  const dataQueries = useMemo(() => {
+    return queries.filter((query) => !isExpressionQuery(query.model));
+  }, [queries]);
+
+  // expression queries only
+  const expressionQueries = useMemo(() => {
+    return queries.filter((query) => isExpressionQueryInAlert(query));
```

Comment:
I needed to add this new type guard function for knowing if the query is `AlertQuery<ExpressionQuery>`